### PR TITLE
Adding in config option for git binary

### PR DESCRIPTION
**Pull request by benrcole**

### CODE REVIEW

File: DataCollector/GitCollector.php [hunk 1]
```diff
@@ -13,17 +14,28 @@ class GitCollector extends DataCollector
...
+    private $gitBinaryPath;
+
     /**
      * Constructor
      *
      * @param string $rootDir
+     * @param string $gitBinaryPath
      */
-    public function __construct($rootDir)
+    public function __construct($rootDir,$gitBinaryPath=null)
```
Comment (jdecool) [inline DataCollector/GitCollector.php]: Please respect codestyling by adding a space after comma (for readability)

File: DataCollector/GitCollector.php [hunk 2]
```diff
@@ -13,17 +14,28 @@ class GitCollector extends DataCollector
...
     {
         $this->rootDir = realpath($rootDir);
         if (false === $this->rootDir) {
             throw new \InvalidArgumentException(sprintf('"%s" is not a valid path.', $rootDir));
         }
+
+        if($gitBinaryPath){
```
Comment (jdecool) [inline DataCollector/GitCollector.php]: This conditional is useless.

Default value is `null` (the same as the default property value).

By removing it, the code depth structure is reduced.

File: DataCollector/GitCollector.php [hunk 3]
```diff
@@ -32,7 +44,11 @@ public function __construct($rootDir)
     public function collect(Request $request, Response $response, \Exception $exception = null)
     {
         try {
-            $repository = new Repository($this->rootDir);
+            $gitBinary=null;
+            if($this->gitBinaryPath){
+                $gitBinary = new GitBinary($this->gitBinaryPath);
+            }
+            $repository = new Repository($this->rootDir,$gitBinary);
```
Comment (jdecool) [inline DataCollector/GitCollector.php]: As the default path of `GitBinary` is set to null, you can remove your conditionnal.

```php
$gitBinary = new GitBinary($this->gitBinaryPath);
$repository = new Repository($this->rootDir, $gitBinary);
```

Comment (jdecool) [inline DataCollector/GitCollector.php]: Please add a space after comma to respect code styling

File: DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -14,6 +14,10 @@ public function getConfigTreeBuilder()
     {
         $treeBuilder = new TreeBuilder();
         $rootNode = $treeBuilder->root('j_decool_git_branch_profiler');
+        $rootNode
+            ->children()
+            ->scalarNode('git_binary_path')->end()
```
Comment (jdecool) [inline DependencyInjection/Configuration.php]: Indentation is missing here

File: DependencyInjection/JDecoolGitProfilerExtension.php [hunk 5]
```diff
@@ -24,5 +24,11 @@ public function load(array $configs, ContainerBuilder $container)
 
         $loader = new Loader\YamlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
         $loader->load('services.yml');
+
+        if(array_key_exists('git_binary_path',$config))
```
Comment (jdecool) [inline DependencyInjection/JDecoolGitProfilerExtension.php]: Add a space after comma to respect codestyling

**Review by jdecool — CHANGES_REQUESTED** [review]
Thanks for your contribution, I made some feedback.

After having resolve it, I will merge your PR.

File: DataCollector/GitCollector.php (hunk 2 above)
Comment (benrcole) [reply DataCollector/GitCollector.php]: The conditional is needed to only trigger `file_exists` when a path has been submitted

**Review by jdecool — APPROVED** [review]

**Comment by jdecool** [comment]
Great thanks for your contribution

**Pull request merged**